src/html/HTMLWriter.jl

```diff
@@ -800,9 +799,34 @@ function render(doc::Documenter.Document, settings::HTML = HTML())
         copy_asset("themes/$(theme).css", doc)
     end
 
-    size_limit_successes = map(collect(keys(doc.blueprint.pages))) do page
+    function page_navnode(page)
```

Comment:
Bikeshed, but let's move this out of the function body perhaps. I am not a huge fan of declaring functions within functions. I think it just needs to take `doc::Document` as the first argument?